modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLayout from '@clayui/layout';
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import TranslationAdminModal from './TranslationAdminModal';
+
+const emptyArray = [];
```

Comment:
It's actually more Javascript 303:

With this pattern if we define the default value inline, every time the component receives a new prop, all the defaults are reset, so when that happen that prop is set with a NEW array and that will cause duplicate renders in the best case and an infinite loop in the worst case scenario.

But if you define the array as a constant outside the component it will always be set with the same value, so it won't render more than once.

Here a sample:

https://codesandbox.io/s/jovial-star-2jrcn?file=/src/App.js

Change 

```
anArray: initialAnArray = emptyArray
```

to

```
anArray: initialAnArray = []
```

to see the magic.

Reply:
Oh, I see. Interesting! I never stumbled upon the issue, because this is a bit of an odd structure. Does it make sense for a prop to be is a trigger for `useEffect`? :thinking:  If I need to set a value of something based on a prop, I would use `useRef`. `useEffect` makes most sense to me as a monitor of state value.

Reply:
But if you want that variable to cause a re-render (which is the case) you need to use state, that's why we don't use refs here.

Reply:
Weird, I've never seen that behavior before until now. 

Only recommendation here would be to add a comment above your `emptyArray` variable declaration specifying why we move it to a variable so that it doesn't get refactored on accident later.

Reply:
@bryceosterhaus we use noop functions because the same reason, which I guess is more common to see in DXP.

I can drop a comment there, for sure, and we should find a way to teach this pattern so we don't need to add a comment every time this happen.